tests/core/scenario/test_scenario.py

```diff
@@ -226,6 +225,51 @@ def get_label(self):
     }
 
 
+def test_raise_pipeline_tasks_not_in_scenario():
+    task_1 = Task("task_1", {}, print)
+    task_2 = Task("task_2", {}, print)
+    with pytest.raises(PipelineTaskDoesNotExistInSameScenario):
+        Scenario("scenario", [], {}, pipelines={"pipeline": {"tasks": [task_1]}})
+
+    with pytest.raises(PipelineTaskDoesNotExistInSameScenario):
+        Scenario("scenario", [task_1], {}, pipelines={"pipeline": {"tasks": [task_1, task_2]}})
+
+    Scenario("scenario", [task_1], {}, pipelines={"pipeline": {"tasks": [task_1]}})
+    Scenario(
+        "scenario",
+        [task_1, task_2],
+        {},
+        pipelines={"pipeline_1": {"tasks": [task_1]}, "pipeline_2": {"tasks": [task_1, task_2]}},
+    )
+
+    scenario = Scenario("scenario", [], {})
+
+    scenario_manager = _ScenarioManagerFactory._build_manager()
+    task_manager = _TaskManagerFactory._build_manager()
+
+    scenario_manager._set(scenario)
+    task_manager._set(task_1)
+    task_manager._set(task_2)
+
+    scenario.add_pipelines({"pipeline_1": {}})
+
+    with pytest.raises(PipelineTaskDoesNotExistInSameScenario):
+        scenario.add_pipelines({"pipeline_2": {"tasks": [task_1]}})
+
+    scenario.tasks = [task_1]
+
+    scenario.add_pipelines({"pipeline_2": {"tasks": [task_1]}})
+
+    with pytest.raises(PipelineTaskDoesNotExistInSameScenario):
+        scenario.add_pipelines({"pipeline_3": {"tasks": [task_2]}})
+
+    with pytest.raises(PipelineTaskDoesNotExistInSameScenario):
+        scenario.add_pipelines({"pipeline_4": {"tasks": [task_1, task_2]}})
```

Comment:
I think we should also check for the message of the Exception.

```suggestion
    with pytest.raises(PipelineTaskDoesNotExistInSameScenario) as error:
        scenario.add_pipelines({"pipeline_3": {"tasks": [task_2]}})
        assert "[task_2] ..." in error.message

    with pytest.raises(PipelineTaskDoesNotExistInSameScenario) as error:
        scenario.add_pipelines({"pipeline_4": {"tasks": [task_1, task_2]}})
        assert "[task_2] ..." in error.message
        assert "task_1" not in error.message
```